src/SprykerMiddleware/Zed/Process/ProcessDependencyProvider.php

```diff
@@ -7,27 +7,55 @@
 class ProcessDependencyProvider extends AbstractBundleDependencyProvider
 {
     const MIDDLEWARE_PROCESS_STAGES = 'MIDDLEWARE_PROCESS_STAGES';
+    const MIDDLEWARE_PROCESS_ITERATORS = 'MIDDLEWARE_PROCESS_ITERATORS';
+    const MIDDLEWARE_PROCESS_LOGGERS = 'MIDDLEWARE_PROCESS_LOGGERS';
+    const MIDDLEWARE_PRE_PROCESSOR_HOOKS_STACK = 'MIDDLEWARE_PRE_PROCESSOR_HOOKS_STACK';
+    const MIDDLEWARE_POST_PROCESSOR_HOOKS_STACK = 'MIDDLEWARE_POST_PROCESSOR_HOOKS_STACK';
 
     /**
      * @param \Spryker\Zed\Kernel\Container $container
      *
      * @return \Spryker\Zed\Kernel\Container
      */
-    public function provideCommunicationLayerDependencies(Container $container)
+    public function provideBusinessLayerDependencies(Container $container)
     {
-        $container = parent::provideCommunicationLayerDependencies($container);
+        $container = parent::provideBusinessLayerDependencies($container);
 
         $container[self::MIDDLEWARE_PROCESS_STAGES] = function () {
             return $this->registerProcessStages();
         };
 
+        $container[self::MIDDLEWARE_PRE_PROCESSOR_HOOKS_STACK] = function () {
+            return $this->registerPreProcessorHooks();
+        };
+
+        $container[self::MIDDLEWARE_POST_PROCESSOR_HOOKS_STACK] = function () {
+            return $this->registerPostProcessorHooks();
+        };
+
         return $container;
     }
 
     /**
      * @return array
      */
-    public function registerProcessStages()
+    public function registerProcessorStages(): array
```

Comment:
we might want to add comments on these methods that describe the expected array structure (`process_name => stack`). what do you think?